rpc/rpcserver.go

```diff
@@ -58,7 +61,9 @@ func (r *rpcServer) Start(ctx context.Context, in *api.StartRequest) (*api.Start
 		return nil, err
 	}
 
-	if err := r.s.Start(b); err != nil {
+	serviceCtx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
```

Comment:
Can we just use the `context` that's passed in instead of the `NotifyContext`?  The caller of `Start` imho should decide when and how it is canceled.

Reply:
No, that's not possible :) The context, it's called in, is a GRPC request. It will be cancelled when the GRPC call finishes. That's not what is needed. Here, I want to create a long-living context for the lifetime of the Service, which is until the application closes (the Service cannot be stopped).

Reply:
Then this feels out of place here. Maybe the `NotifyContext` should then be created within `r.s.Start`?

Reply:
Refactored, now SIGINT is handled at a single place in server/server.go. The entity spawning a Service is directly responsible for calling `Service::Shutdown()`. I followed the design of `http.Server` from the stdlib here.